### PR TITLE
Add CUDA 11.6 images

### PR DESCRIPTION
**Pull request by jakirkham**

### CODE REVIEW

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Isuru! 😄

@kkraus14 @jaimergp, could you please also take a look?

**Review by jaimergp — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -106,6 +106,11 @@ jobs:
           CUDA_VER: "11.5.1"
           CENTOS_VER: "8"
 
+        - DOCKERIMAGE: linux-anvil-cuda
+          DOCKERTAG: "11.6"
+          CUDA_VER: "11.6.0"
+          CENTOS_VER: "8"
```
Comment (kkraus14) [inline .github/workflows/ci.yaml:112]: Is there a reason we moved to CentOS 8 for CUDA 11.6 / 11.5 on x86? CentOS 7 is still supported as far as I can tell?

**Review by kkraus14 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jakirkham) [reply .github/workflows/ci.yaml:112]: Interesting I seem to recall CentOS 7 wasn't supported here before and recall us discussing this, but can't find that atm. Asking here ( https://github.com/conda-forge/docker-images/pull/198#discussion_r787116632 )

Comment (jakirkham) [reply .github/workflows/ci.yaml:112]: cc @ajschmidt8 @leofang

Comment (leofang) [reply .github/workflows/ci.yaml:112]: @jakirkham This is a question we should ask the image providers internally. If you search on the docket hub, there's no cos7 based image for CUDA 11.5.1, but there is one for 11.6.0.

Comment (jakirkham) [reply .github/workflows/ci.yaml:112]: Please see this issue ( https://gitlab.com/nvidia/container-images/cuda/-/issues/145 )

Comment (kkraus14) [reply .github/workflows/ci.yaml:112]: So should we use CentOS 7 for CUDA 11.6?

Comment (jakirkham) [reply .github/workflows/ci.yaml:112]: Checked with others and CentOS 7 should be supported by both. We just lack an image for 11.5 atm.

Yeah switching this to CentOS 7 makes sense. Would also suggest switching the 11.5 images to CentOS 7 once available. Have raised issue ( https://github.com/conda-forge/docker-images/issues/205 ) on this point. Am also subscribed to the upstream issue.

Comment (jakirkham) [reply .github/workflows/ci.yaml]:
```suggestion
          CENTOS_VER: "7"
```

**Review by leofang — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**